Pharmacy/PharamcyApi/Controllers/MedicineProcurementController.cs

```diff
@@ -23,24 +24,46 @@ public MedicineProcurementController(IUnitOfWork uow)
         }
 
         [HttpPost("check")]
-        public IActionResult CheckMedicineAvailability(CheckMedicineAvailabilityRequestDTO checkMedicineAvailabilityRequestDTO)
+        public IActionResult CheckMedicineAvailability(CheckMedicineAvailabilityRequestDTO requestDTO)
         {
             try
             {
-                _hospitalAuthService.ValidateApiKey(checkMedicineAvailabilityRequestDTO.ApiKey);
-                bool isMedicineAvailable = _medicineProcurementService
-                    .IsMedicineAvailable(checkMedicineAvailabilityRequestDTO.MedicineName,
-                    checkMedicineAvailabilityRequestDTO.ManufacturerName, checkMedicineAvailabilityRequestDTO.Quantity);
+                _hospitalAuthService.ValidateApiKey(requestDTO.ApiKey);
+                bool answer = _medicineProcurementService.IsMedicineAvailable(requestDTO.MedicineName, requestDTO.ManufacturerName, requestDTO.Quantity);
 
-                return Ok(new CheckMedicineAvailabilityResponseDTO() { Answer = isMedicineAvailable });
+                return Ok(new CheckMedicineAvailabilityResponseDTO() { Answer = answer });
             }
-            catch (UnauthorizedAccessException uae)
+            catch (UnauthorizedAccessException exception)
             {
-                return Unauthorized(uae.Message);
+                return Unauthorized(exception.Message);
             } 
-            catch (KeyNotFoundException knfe)
+            catch (MedicineFromManufacturerNotFoundException exception)
             {
-                return NotFound(knfe.Message);
+                return NotFound(exception.Message);
+            }
+        }
+
+        [HttpPost("execute")]
+        public IActionResult ExecuteMedicineProcurement(MedicineProcurementRequestDTO requestDTO)
+        {
+            try
+            {
+                _hospitalAuthService.ValidateApiKey(requestDTO.ApiKey);
+                _medicineProcurementService.ExecuteProcurement(requestDTO.MedicineName, requestDTO.ManufacturerName, requestDTO.Quantity);
+
+                return Ok("Procurement executed successfully.");
+            }
+            catch (UnauthorizedAccessException exception)
+            {
+                return Unauthorized(exception.Message);
+            }
```

Comment:
Since these two have exactly the same outcome, why don't we make them a little bit more flexible and throw the same exception with a different message?